upb/def.c

```diff
@@ -1018,14 +1018,20 @@ static int field_number_cmp(const void *p1, const void *p2) {
   return f1->number - f2->number;
 }
 
-static void assign_layout_indices(const upb_msgdef *m, upb_msglayout_field *fields) {
+static void assign_layout_indices(const upb_msgdef *m, upb_msglayout *l,
+                                  upb_msglayout_field *fields) {
   int i;
   int n = upb_msgdef_numfields(m);
+  int dense_below = 0;
   for (i = 0; i < n; i++) {
     upb_fielddef *f = (upb_fielddef*)upb_msgdef_itof(m, fields[i].number);
     UPB_ASSERT(f);
     f->layout_index = i;
+    if (i <= 254 && upb_fielddef_number(f) == i + 1) {
```

Comment:
this uses 254, but the version below using uint8_max which is 256.  Which should it be?

Reply:
`UINT8_MAX` is 255 (0xff). If the predicate is true, we will assign `i + 1`, so `i <= 254` is the correct check I believe.

I wrote this predicate slightly differently from the one in the code generator because this one is size/speed-sensitive. I did the range check against `i` instead of the field number, because `i` is the loop index, which makes it easier for the compiler to reason about, and if `i > 254` we can avoid the load of `f->number` altogether.

Reply:
I changed the logic to use `UINT8_MAX` as the constant we compare against.